easybuild/easyblocks/i/icc.py

```diff
@@ -105,53 +117,117 @@ def sanity_check_step(self):
+                    debuggerpath = os.path.join('composer_xe_%s' % self.version.split('.')[0], 'debugger')
+
+                libpaths = [
+                    os.path.join('tbb/lib/intel64', tbbgccversion),
+                    'ipp/lib/intel64',
+                    'debugger/ipt/intel64/lib',
+                    'lib/intel64',
+                    'compiler/lib/intel64',
+                ]
+                dirmap = {
+                    'PATH': ['debugger/gdb/intel64/bin', 'ipp/bin/intel64', 'tbb/bin/intel64', 'bin/intel64'],
+                    'LD_LIBRARY_PATH': libpaths,
+                    'LIBRARY_PATH': libpaths,
+                    'MANPATH': ['debugger/gdb/intel64/share/man', 'man/en_US', 'share/man', 'man'],
+                    'CPATH': ['ipp/include', 'tbb/include'],
+                }
+            else:
+                # New Directory Layout for Intel Parallel Studio XE 2016
+                # https://software.intel.com/en-us/articles/new-directory-layout-for-intel-parallel-studio-xe-2016
+                prefix = "compilers_and_libraries_%s/linux" % self.version
+                # Debugger requires INTEL_PYTHONHOME, which only allows for a single value
+                debuggerpath = 'debugger_%s' % self.version.split('.')[0]
+
+                libpaths = [
+                    'daal/../compiler/lib/intel64_lin',
+                    os.path.join('daal/../tbb/lib/intel64_lin', tbbgccversion),
+                    'daal/lib/intel64_lin',
+                    os.path.join(debuggerpath, 'libipt/intel64/lib'),
+                    os.path.join('tbb/lib/intel64', tbbgccversion),
+                    'mkl/lib/intel64',
+                    'ipp/lib/intel64',
+                    'ipp/../compiler/lib/intel64',
+                    'mpi/intel64',
+                    'compiler/lib/intel64',
+                    'lib/intel64_lin',
+                ]
+                dirmap = {
+                    'PATH': [
+                        'mpi/intel64/bin',
+                        'ipp/bin/intel64',
+                        os.path.join(debuggerpath, 'gdb/intel64/bin'),
+                        'bin/intel64',
+                        'bin',
+                    ],
+                    'LD_LIBRARY_PATH': libpaths,
+                    'LIBRARY_PATH': libpaths,
+                    'MANPATH': ['man/common', 'man/en_US', 'debugger/gdb/intel64/share/man'],
+                    'CPATH': ['ipp/include', 'mkl/include', 'tbb/include', 'daal/include'],
+                    'DAALROOT': ['daal'],
+                    'TBBROOT': ['tbb'],
+                    'IPPROOT': ['ipp'],
+                    'CLASSPATH': ['daal/lib/daal.jar']
+                }
+
+        # set debugger path
+        if debuggerpath:
+            if os.path.isdir(os.path.join(self.installdir, debuggerpath, 'python/intel64')):
+                self.cfg['modextravars'] = {'INTEL_PYTHONHOME': os.path.join('$root', debuggerpath, 'python/intel64')}
```

Comment:
don't hard modify self.cfg['modextravars'], there may be something there (from the easyconfig file)

better update `dirmap` so that it also specifies `INTEL_PYTHONHOME`

but strictly speaking, this should be handled via `make_module_extra`